### ISSUE TITLE
[BUG] VM pods turn to the terminating state after switching the VLAN interface

### ISSUE THREAD

**Issue by yaocw2020** [harvester/harvester#1464]
**Describe the bug**

VM pods turn to the terminating state because of downtime after switching the VLAN interface.

**To Reproduce**
1. Setup a VM in a certain host.
2. Switch the VLAN interface of this host.

**Expected behavior**

**Support bundle**

**Environment:**
 - Harvester ISO version:  0.3
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): Baremetal with Dell PowerEdge R630

**Additional context**

**Comment by yaocw2020** [comment]
```
bm-harv-42:/home/rancher # kubectl get pod -o wide
NAME                                            READY   STATUS        RESTARTS   AGE    IP            NODE          NOMINATED NODE   READINESS GATES
virt-launcher-lawr-h-g57m9                      0/1     Terminating   0          5h1m   10.52.1.52    bm-harv-42    <none>           <none>
virt-launcher-michelia-pk2j8                    0/1     Terminating   0          23h    10.52.1.232   bm-harv-42    <none>           <none>
virt-launcher-opensuse-02-vlv82                 0/1     Terminating   0          41h    10.52.1.184   bm-harv-42    <none>           <none>
virt-launcher-opensuse-178-2-7977f              0/1     Terminating   0          106m   10.52.1.55    bm-harv-42    <none>           <none>
virt-launcher-opensuse188-vhqct                 0/1     Terminating   0          22h    10.52.1.45    bm-harv-42    <none>           <none>
virt-launcher-opensuse3-rzgtn                   0/1     Terminating   0          24h    10.52.1.212   bm-harv-42    <none>           <none>
virt-launcher-rke2-pool1-ce1c740f-hjcm9-j4m5m   1/1     Running       0          164m   10.52.0.235   bm-harv-4-2   <none>           <none>
virt-launcher-test-schedule-qpmdm               0/1     Terminating   0          24h    10.52.1.215   bm-harv-42    <none>           <none>
virt-launcher-wj2-mwkhk                         1/1     Running       0          70m    10.52.2.85    bm-harv-45    <none>           <none>
wordpress-mysql-6965fc8cc8-6rgqn                1/1     Running       0          3d2h   10.52.2.250   bm-harv-45    <none>           <none>
```
```
bm-harv-42:/home/rancher # kubectl describe pod virt-launcher-lawr-h-g57m9
......
Events:
  Type     Reason        Age   From             Message
  ----     ------        ----  ----             -------
  Warning  NodeNotReady  14m   node-controller  Node is not ready
  Normal   Killing       12m   kubelet          Stopping container compute
```

**Comment by yaocw2020** [comment]
For testing: 
Same as https://github.com/harvester/harvester/issues/1529.
The VM pod should be normal when disabling the VLAN network or switching the network interface for the VLAN network.

**Comment by TachunLin** [comment]
Verified **fixed** on master-1e042e04-head. Close this issue.

### Result
Switch the VLAN interface of this host can't affect Host and VM operation. 
Both keep in `running` status. 

![image](https://user-images.githubusercontent.com/29251855/144206164-092272aa-0488-40f4-bb3d-4a1aea5fdb5d.png)

![image](https://user-images.githubusercontent.com/29251855/144207427-79056433-94d6-4d2f-9680-5a55a9d80efe.png)

### Test Information
* Test Environment: 3 nodes harvester on local kvm machines
* Harvester version: master-1e042e04-head

### Verify Steps
1. User ipxe-example to build up 3 nodes harvester 
2. Login harvester dashboard -> Access Settings 
3. Enable vlan network with `harvester-mgmt` NIC interface
4. Create a VM using `harvester-mgmt`
5. Disable vlan network
6. Enable vlan network and select `bond0` interface
![image](https://user-images.githubusercontent.com/29251855/144204800-ed20ab79-0c18-4a70-b258-2468d62e072d.png)
7. Check host and vm is working 
8. Directly switch network interface from `bond0` to `harvester-mgmt` without disable it.
![image](https://user-images.githubusercontent.com/29251855/144206080-cbba3e29-b125-422a-b629-9a412a218feb.png)
9. Check host and vm is working
